wit/runtime_config.wit

```diff
@@ -0,0 +1,10 @@
+package wasi:config@0.1.0;
+
+interface runtime-config {
+    use types.{config-error};
+
+    /// Gets a single opaque config value set at the given key if it exists
+    get: func(key: string) -> result<option<list<u8>>, config-error>;
+    /// Gets a list of all set config data
+    get-all: func() -> result<list<tuple<string, list<u8>>>, config-error>;
```

Comment:
What are some use cases for `get-all`?

Reply:
Some apps and libraries iterate over environment variables to look for prefix matches, e.g. "find all variables starting with FOO_BAR_" as a way of encoding a variable-length map or list. So perhaps this would make it easier to port such software.

Reply:
Yeah @dicej had it right. There is also a use case where you want to fetch all the config once and then iterate over it in memory rather than doing individual fetches